### PR TITLE
Junit5 test with dependencies runner

### PR DESCRIPTION
**Pull request by JonasKunz**
## What does this PR do?

Adds a new `TestClassWithDependenciesRunner` variant which supports JUnit5 test classes.
Also did a bit of refactoring and moved the runner into the existing `testutils` package.

## Checklist

- [x] This is something else
  - [ ] ~I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)~

### CODE REVIEW

**Review by SylvainJuge — APPROVED** [review]
:tada: Good job! this unblocks the Junit5 migration that we aim to do for a while.

File: apm-agent-core/src/test/java/co/elastic/apm/agent/testutils/TestClassWithDependencyRunner.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    public static class DisableOutsideOfRunnerCondition implements ExecutionCondition {
+
+        @Override
+        public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext context) {
+            AnnotatedElement element = context.getElement().orElse(null);
+            return findAnnotation(element, DisableOutsideOfRunner.class)
+                .map(annotation -> disabled(element + " is @DisableOutsideDependencyRunner", annotation.value()))
+                .orElse(enabled("@DisableOutsideDependencyRunner is not present"));
+        }
+    }
```
[the remark this answers is not in the document]
Comment (JonasKunz) [reply apm-agent-core/src/test/java/co/elastic/apm/agent/testutils/TestClassWithDependencyRunner.java:72]: This is useful if you want to write integration tests which ensure that your instrumentation is **not** active if the target library to instrument is too old.
See for example [this test](https://github.com/JonasKunz/apm-agent-java/blob/otel-metricsdk-instrumentation/apm-agent-plugins/apm-opentelemetry/apm-opentelemetry-metricsdk-plugin/src/test/java/co/elastic/apm/agent/otelmetricsdk/UnsupportedSdkVersionIgnoredTest.java): If the annotation was not present, the test would also be executed outside of the integration tests with the latest otel sdk version. The latest otel sdk version however is supported, therefore the test would always fail.

[the remark this answers is not in the document]
Comment (SylvainJuge) [reply apm-agent-core/src/test/java/co/elastic/apm/agent/testutils/TestClassWithDependencyRunner.java:72]: :+1:, do you think you could add this to the annotation javadoc to make sure I don't ask this again in the future ?

**Pull request merged**